lib/packages/docutils/rst.nim

```diff
@@ -579,6 +582,23 @@ proc whichRoleAux(sym: string): RstNodeKind =
   else:  # unknown role
     result = rnUnknownRole
 
+proc setCurrFilename(s: PRstSharedState, file1: string) =
+  for i, file2 in s.files:
+    if file1 == file2:
+      s.currFileIdx = i.FileIndex
+      return
+  s.files.add file1
+  s.currFileIdx = (s.files.len - 1).FileIndex
+
+proc getFilename(files: seq[string], fid: FileIndex): string =
+  if fid.int < files.len:
+    result = files[fid.int]
+  else:
+    result = "input"
```

Comment:
what is `input` used for? is that same as `compiler/msgs.nim:177:3:  commandLineDesc* = "command line"` ?

Reply:
just a fallback value, it should not happen. Alternatively a ValueError can be raised here.

Reply:
if it shouldn't happen than maybe use doAssert? (ie indicates program logic bug, not user error)

Reply:
OK, switched to `doAssert`.

Reply:
... and doAssert caught a bug :-)